python/fastscapelib/eroders/numba_eroder.py

```diff
@@ -0,0 +1,71 @@
+import inspect
+
+from fastscapelib.flow import create_flow_kernel
+
+
+class NumbaEroderType(type):
+    @classmethod
+    def check_errs(cls, dct):
+        errs = []
+        missing_members = {"spec", "outputs", "application_order"}.difference(
+            set(dct.keys())
+        )
+        if missing_members:
+            errs.append(f"Missing mandatory class members {missing_members}")
+
+        if "eroder_kernel" not in dct:
+            errs.append(f" Missing mandatory class method 'eroder_kernel'")
+        elif not isinstance(dct["eroder_kernel"], staticmethod):
+            errs.append(
+                "Method 'eroder_kernel' must be static (add the '@staticmethod' decorator)"
+            )
+        else:
+            kernel_sig = inspect.signature(dct["eroder_kernel"])
+            if len(kernel_sig.parameters) != 1:
+                errs.append(
+                    "Method 'eroder_kernel' must take a single argument (the node data)"
+                )
+
+        return errs
+
+    def __new__(cls, name, bases, dct):
+        if bases:
+            errs = cls.check_errs(dct)
+            if errs:
+                err_msg = "\n  - ".join(errs)
+                raise TypeError(f"Can't instantiate class {name}:\n  - {err_msg}")
+
+        instance = super().__new__(cls, name, bases, dct)
+        return instance
+
+
+class NumbaEroderBase(metaclass=NumbaEroderType):
```

Comment:
I think I would prefer a class decorator over a base class here. A decorator would be:

- similar to Numba's `jitclass` decorator
- a bit better for documenting the required members `spec`, `outputs` and `application_order`
- also similar to how we use the fastscape high-level API (via xarray-simlab "process" decorated classes)


This can be addressed in a follow-up PR, though.